api/maven-api-xml/src/main/java/org/apache/maven/api/xml/XmlNode.java

```diff
@@ -18,121 +18,244 @@
 
-    String SELF_COMBINATION_REMOVE = "remove";
+    public static final String SELF_COMBINATION_REMOVE = "remove";
 
     /**
      * In case of complex XML structures, combining can be done based on id.
      */
-    String ID_COMBINATION_MODE_ATTRIBUTE = "combine.id";
+    public static final String ID_COMBINATION_MODE_ATTRIBUTE = "combine.id";
 
     /**
      * In case of complex XML structures, combining can be done based on keys.
      * This is a comma separated list of attribute names.
      */
-    String KEYS_COMBINATION_MODE_ATTRIBUTE = "combine.keys";
+    public static final String KEYS_COMBINATION_MODE_ATTRIBUTE = "combine.keys";
 
     /**
      * This default mode for combining a DOM node during merge means that where element names match, the process will
      * try to merge the element attributes and values, rather than overriding the recessive element completely with the
      * dominant one. This means that wherever the dominant element doesn't provide the value or a particular attribute,
      * that value or attribute will be set from the recessive DOM node.
      */
-    String DEFAULT_SELF_COMBINATION_MODE = SELF_COMBINATION_MERGE;
+    public static final String DEFAULT_SELF_COMBINATION_MODE = SELF_COMBINATION_MERGE;
+
+    protected final String prefix;
+
+    protected final String namespaceUri;
+
+    protected final String name;
+
+    protected final String value;
+
+    protected final Map<String, String> attributes;
+
+    protected final List<XmlNode> children;
+
+    protected final Object location;
+
+    public XmlNode(String name) {
+        this(name, null, null, null, null);
+    }
+
+    public XmlNode(String name, String value) {
+        this(name, value, null, null, null);
+    }
+
+    public XmlNode(XmlNode from, String name) {
+        this(name, from.getValue(), from.getAttributes(), from.getChildren(), from.getInputLocation());
+    }
+
+    public XmlNode(String name, String value, Map<String, String> attributes, List<XmlNode> children, Object location) {
+        this("", "", name, value, attributes, children, location);
+    }
+
+    public XmlNode(
+            String prefix,
+            String namespaceUri,
+            String name,
+            String value,
+            Map<String, String> attributes,
+            List<XmlNode> children,
+            Object location) {
+        this.prefix = prefix == null ? "" : prefix;
+        this.namespaceUri = namespaceUri == null ? "" : namespaceUri;
+        this.name = Objects.requireNonNull(name);
+        this.value = value;
+        this.attributes = ImmutableCollections.copy(attributes);
+        this.children = ImmutableCollections.copy(children);
+        this.location = location;
+    }
+    // ----------------------------------------------------------------------
+    // Name handling
+    // ----------------------------------------------------------------------
+
+    public String getPrefix() {
```

Comment:
Javadoc on all public methods, and thi9s should be more than simply "Returns prefix". It should say what a prefix is.